Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -192,43 +191,28 @@ FwuTopSwapSetting (
     pFwUpdStatus = (FW_UPDATE_STATUS *)(UINTN)RsvdBase;
   }
 
-  //
-  // Get Top swap register Bit0 in PCH Private Configuration Space.
-  //
-  P2sbBase   = MM_PCI_ADDRESS (0, PCI_DEVICE_NUMBER_PCH_LPC, 1, 0); // P2SB device base
-
-  if (MmioRead16 (P2sbBase) == 0xFFFF) {
-    //
-    // unhide P2SB
-    //
-    MmioWrite8 (P2sbBase + 0xE1, 0);
-    DEBUG ((DEBUG_INFO, "P2sb is hidden, unhide it\n"));
+  // If in a recovery path, stay on current partition
+  if (PcdGetBool (PcdSblResiliencyEnabled) &&
+      (GetFailedBootCount () >= PcdGet8 (PcdBootFailureThreshold) ||
+      pFwUpdStatus->StateMachine == FW_UPDATE_SM_RECOVERY)) {
+    return;
   }
 
-  P2sbBar    = MmioRead32 (P2sbBase + 0x10);
-  P2sbBar  &= 0xFFFFFFF0;
-  ASSERT (P2sbBar != 0xFFFFFFF0);
```

Comment:
Is this redundant code?

Reply:
This code is only here so that the contents of the TS register can be printed in a debug statement. I didn't find this debug statement to be particularly useful in the context of this function, so I got rid of it. The boot partition indicated by this register is incorporated in other debug statements.